lib/python/qmk/cli/__init__.py

```diff
@@ -49,3 +96,48 @@
 if sys.version_info[0] != 3 or sys.version_info[1] < 7:
     cli.log.error('Your Python is too old! Please upgrade to Python 3.7 or later.')
     exit(127)
+
+# Check to make sure we have all our dependencies
+msg_install = 'Please run `python3 -m pip install -r %s` to install required python dependencies.'
+
+if _broken_module_imports('requirements.txt'):
+    if yesno('Would you like to install the required Python modules?'):
+        _run_cmd(sys.executable, '-m', 'pip', 'install', '-r', 'requirements.txt')
+    else:
+        print()
+        print(msg_install % (str(Path('requirements.txt').resolve()),))
+        print()
+        exit(1)
+
+if cli.config.user.developer and _broken_module_imports('requirements-dev.txt'):
+    if yesno('Would you like to install the required developer Python modules?'):
+        _run_cmd(sys.executable, '-m', 'pip', 'install', '-r', 'requirements.txt')
+    else:
+        print()
+        print(msg_install % (str(Path('requirements-dev.txt').resolve()),))
+        print('You can also turn off developer mode: qmk config user.developer=None')
+        print()
+        exit(1)
```

Comment:
```suggestion
if cli.config.user.developer:
    args = sys.argv[1:]
    while args and args[0][0] == '-':
        del args[0]
    if not args or args[0] != 'config':
        if _broken_module_imports('requirements-dev.txt'):
            if yesno('Would you like to install the required developer Python modules?'):
                _run_cmd(sys.executable, '-m', 'pip', 'install', '-r', 'requirements-dev.txt')
            else:
                print()
                print(msg_install % (str(Path('requirements-dev.txt').resolve()),))
                print('You can also turn off developer mode: qmk config user.developer=None')
                print()
                exit(1)
```
We want to allow the user to run `qmk config` so they can turn off dev mode.